synapseclient/client.py

```diff
@@ -1657,7 +1678,7 @@ def store(
                     if (synapseStore or local_state_fh.get("externalURL") is None)
                     else local_state_fh.get("externalURL"),
                     synapseStore=synapseStore,
-                    md5=local_state_fh.get("contentMd5"),
+                    md5=local_file_md5_hex or local_state_fh.get("contentMd5"),
```

Comment:
What happens if both these things are `None`?

Reply:
Further down in the code it will get the md5 from the file on disk. Both are none during the first upload of a file - since the code above is only checked for a file that already exists in Synapse. And the file handle hasn't been created yet.